tests/test_livysession.py

```diff
@@ -436,6 +438,49 @@ def test_create_sql_hive_context_happens_once(self):
                in http_client.post_statement.call_args_list
         assert len(http_client.post_statement.call_args_list) == 1
 
+        session._get_sql_context_creation_command = MagicMock()
```

Comment:
This test is massive now. Can you please break it up into different tests. The current test should only `test_create_sql_hive_context_happens_once`. This test is now testing what happens on failure.